source/onednngraph.js

```diff
@@ -0,0 +1,515 @@
+                                array = null;
+                            }
+                            else if (array != null) {
+                                array.push(number);
+                            }
+                        }
+                        if (array != null) {
+                            this._value = array;
+                        }
+                    }
+                    break;
+                default:
+                    throw new onednngraph.Error("Unsupported attribute type '" + schema.type + "'.");
+            }
+        }
+    }
+
+    get name() {
+        return this._name;
+    }
+
+    get type() {
+        return this._type;
+    }
+
+    get value() {
+        return this._value;
+    }
+
+    get visible() {
+        return this._visible == false ? false : true;
+    }
+};
+
+onednngraph.Parameter = class {
+
+    constructor(name, args) {
+        this._name = name;
+        this._arguments = args;
+    }
+
+    get name() {
+        return this._name;
+    }
+
+    get visible() {
+        return true;
+    }
+
+    get arguments() {
+        return this._arguments;
+    }
+};
+
+onednngraph.Argument = class {
+
+    constructor(name, type, initializer, lt) {
+        if (typeof name !== 'string') {
+            throw new onednngraph.Error("Invalid argument identifier '" + JSON.stringify(name) + "'.");
+        }
+        this._name = name;
+        this._type = type || null;
+        this._initializer = initializer || null;
+        this._lt = lt;
+    }
+
+    get name() {
+        return this._name;
+    }
+
+    get type() {
+        if (this._initializer) {
+            return this._initializer.type;
+        }
+        return this._type;
+    }
+
+    get initializer() {
+        return this._initializer;
+    }
+
+    get description() {
+        const stride = this._lt.stride ? ('[' + this._lt.stride.map((stride) => stride ? stride.toString() : '?').join(',') + ']') : '';
+        const layout_type = this._lt.layout_type;
+        const property_type = this._lt.property_type;
+        return 'stride = ' + stride + ', layout_type = ' + layout_type + ', property_type = ' + property_type;
+    }
+};
+
+onednngraph.TensorType = class {
+
+    constructor(dataType, shape) {
+        this._dataType = dataType || '?';
+        this._shape = shape;
+    }
+
+    get dataType() {
+        return this._dataType;
+    }
+
+    get shape() {
+        return this._shape;
+    }
+
+    toString() {
+        return this._dataType + this._shape.toString();
+    }
+};
+
+onednngraph.TensorShape = class {
+
+    constructor(dimensions) {
+        this._dimensions = dimensions;
+    }
+
+    get dimensions() {
+        return this._dimensions;
+    }
+
+    toString() {
+        return this._dimensions ? ('[' + this._dimensions.map((dimension) => dimension ? dimension.toString() : '?').join(',') + ']') : '';
+    }
+};
+
+onednngraph.LogicalTensor = class {
```

Comment:
`onednn.Tensor`. Why makes a tensor logical?

Reply:
removed. Actually, it is a simple wraper for meta info of onednn graph tensor. These info can be directly feed into TensorShape and TensorType.

Reply:
`onednn.Tensor` will be needed to show the initial value of weights. For example, `Convolution.inputs[1]` should have a weights value that is exposed as `onednn.Argument.initializer`.

Reply:
For example, "Wildcard"-style nodes in other formats get rendered as part of the `argument.initializer` in other formats. This requires the `initializer` to be set even when empty. It might also help to differentiate between `constant` and `variable` nodes.

Reply:
Great. One question here is that does every inputs of operator need to set initializer? Maybe I need to take a deep understand this part since I missed this previously. BTW, is it possible to support the basic part(graph, node, attr, TensorShape, TensorType) first and then I'll enhance the Tensor part. What do you think?

Reply:
Only inputs that have an initial value, usually the weights. ONNX [example](https://netron.app/?url=https://media.githubusercontent.com/media/onnx/models/main/vision/classification/squeezenet/model/squeezenet1.0-3.onnx) for reference.

Reply:
Thanks for the example. I'll try to fullfill this part according to this.

Reply:
Hi, thanks for your guide. After tried locally, I tend to not set the initializer and also removed the `layout_type` and `property_type` from description. Here are some reasons:
1) onednn graph is a lower level graph compared with framework. It is hard to represent a real `Model` with the graph. For example, the graph doesn't define any `data` or `weight`, and we can only know how many inputs/outputs for each node.
2) Every inputs/outputs of node have `property_type`, which means even for an activation, it can be constant. I've tried to set initializer for every arguments and it appears that edge info all missed.
3) I agree with you that `description` is not a suitable place to show the info. So, I decide to remove them.

Reply:
Sounds good.

1) The input parameter names for types like `Convolution` should be descriptive, (e.g. `input` and `weight` instead of `input0` and `input1`.

2) All other formats fold constant nodes that are used only once to be initializers to make the graph more readable. For example, `Wildcard` nodes with no inputs, a single output and not attributes could be folded into the parent node as an initializer with `category` = `Wildcard`.

Reply:
done.